### ISSUE TITLE
Component: move into selection + add named components

### ISSUE THREAD

**Issue by jmsmyth** [ocadotechnology/hexagonjs#244]
Typically the component api is used like this:

``` es6
const api = {
  ...
}

// to set
hx.component.register(hx.select(selector).node(), api) 

// to get
hx.component(selector)
hx.select(selector).component() // alternative
```

We could move the setting part of the component api into the selection module, so that the api becomes symmetric like everything else:

``` es6
const api = {
  ...
}

// to set
selection.component(api)

// to get
selection.component()
```

Also when doing this, could also introduce named components:

``` es6
selection.component('dropdown', api)  // to set
selection.component('dropdown')       // to get
```

Then `hx.component.register` and `hx.component` could be deprecated, or `hx.component` could be made symmetric.

**Comment by jmsmyth** [comment]
This has been done in the webpack branch. (Called `Selection::api`). It may be worth backporting it to 1.x.x so that the deprecation warnings can be added

**Comment by c-frater** [comment]
Pretty sure you already back-ported this? Selection::api was added 5 months ago: https://github.com/ocadotechnology/hexagonjs/commit/3ec90f0d4fcb90a4f656067ce41d0c43dd76c93e

The only bit that still needs doing is making Selection::component circular (or deprecating/removing it)

**Comment by jmsmyth** [comment]
Ah yeah, just a deprecation notice then. The new version is self contained (and the component module has gone): https://github.com/ocadotechnology/hexagonjs/blob/webpack/modules/selection/main/index.coffee#L452

The named components part is missing from the version in 1.x.x, but that's probably fine

**Comment by c-frater** [comment]
The component methods are deprecated in *2.0.0* with the `api` methods as a replacement.

```
const api = {
  ...
}

// to set
selection
  .api('dropdown', api) // set named API
  .api(api) // set unnamed (default) API

// to get
selection.api() // returns the last set unnamed API
selection.api('dropdown') // Returns the named API
```

The recommendation is to use a named API for explicitness and to cope with the cases where multiple APIs are added to the same selection (e.g. in the case of `Autocomplete` where a `Menu` and `Dropdown` are present on the selection)

Marking this ticket as completed (resolved in #525) - The deprecated methods will be removed in #487
